cli/src/commands/run.rs

```diff
@@ -53,17 +303,84 @@ pub fn cmd_run(ui: &mut Ui, command: &CommandHelper, args: &RunArgs) -> Result<(
     let _resolved_commits: Vec<_> = workspace_command
         .parse_union_revsets(&args.revisions)?
         .evaluate_to_commits()?
+    let resolved_commits: Vec<_> = workspace_command
+        .parse_revset(&args.revisions)?
+        .evaluate_to_commits()?
         .try_collect()?;
     // Jobs are resolved in this order:
     // 1. Commandline argument iff > 0.
     // 2. the amount of cores available.
     // 3. a single job, if all of the above fails.
-    let _jobs = match args.jobs {
+    let jobs = match args.jobs {
         Some(0) => return Err(user_error("must pass at least one job")),
         Some(jobs) => Some(jobs),
         None => std::thread::available_parallelism().map(|t| t.into()).ok(),
     }
     // Fallback to a single user-visible job.
     .unwrap_or(1usize);
+
+    let mut done_commits = HashSet::new();
```

Comment:
Does this make sense to track the done commits for the UI?

Reply:
It seems fine; as opposed to what?

Reply:
any other variant, since we want to integrate this later with [superconsole](../../../../facebookincubator/superconsole)?

Reply:
> One of the things I would like to see here is an extension point for custom output/status aggregation.

Ok, I'll add a TODO. I assume this is meant to stream the output potentially across the network, if I understand you correctly?

Reply:
I've added the TODO.